data/virt_autotest/fetch_logs_from_guest.sh

```diff
@@ -0,0 +1,260 @@
+
+       local guest_devices=`virt-filesystems -d ${guest_domain} | grep -ioE "^/dev.*[^@].*$"`
+       local guest_device=""
+       for onedevice in ${guest_devices[@]};do
+	   (guestfish -r -d ${guest_domain} -m ${onedevice} ls / | grep -ioE "^${guest_filesystem}$") &> /dev/null
+           if [[ $? -eq 0 ]];then
+              guest_device=${onedevice}
+              break
+           else
+              continue
+           fi
+       done
+
+       if [[ ${guest_device} != "" ]];then
+	  echo ${guest_device}
+       else
+	  echo "Can not find disk device that hosts ${guest_filesystem}."
+       fi
+}
+
+#Fetach logs from virtual machine to local host via ssh
+function fetch_logs_from_guest_via_ssh() {
+        local guest_domain=$1
+	local guest_ipaddr=$2
+        local logs_folder=$3
+        local guest_user="root"
+        local guest_pass="novell"
+        local sshpass_scp_cmd="sshpass -p ${guest_pass} scp -r ${guest_user}@${guest_ipaddr}"
+        local sshpass_ssh_cmd="sshpass -p ${guest_pass} ssh ${guest_user}@${guest_ipaddr}"
+	local guest_transformed=${guest_domain//./_}
+
+        local ret_result=128
+        local retry_times=0
+        while [[ ${retry_times} -lt 3 ]] && [[ ${ret_result} -ne 0 ]];
+        do
+	      mkdir -p ${logs_folder}/${guest_transformed}
+              echo -e "${sshpass_scp_cmd}:${logs_folder} ${logs_folder}/${guest_transformed}"
+              ${sshpass_scp_cmd}:${logs_folder} ${logs_folder}/${guest_transformed}
+	      ret_result=$?
+              if [[ ${ret_result} -eq 0 ]];then
+                 echo -e "Successfully fetched ${logs_folder} from guest ${guest_domain} via ssh."
+                 ${sshpass_ssh_cmd} rm -f -r ${logs_folder}          
+		 break
+              fi
+              retry_times=$((${retry_times}+1))
+        done
+
+	if [[ ${ret_result} -ne 0 ]];then
+            echo -e "Failed to fetch ${logs_folder} from guest ${guest_domain} via ssh."
+        fi
+
+	return ${ret_result}
+}
+
+#Fetch logs from virtual machine to local host by using libguestfs tools
+function fetch_logs_from_guest_via_libguestfS() {
+        local guest_domain=$1
+	local logs_fetched=$2
+	local logs_folder=$3
+
+        setup_libguestfs_env
+	virt-filesystems -d ${guest_domain} &> /dev/null
+	if [[ $? -ne 0 ]];then
+	   echo -e "Running ${guest_domain} can not be accessed by libguestfs currently. Shut it down now."
+	   virsh destroy ${guest_domain}
+	fi
+	local guest_device=`find_disk_hosts_filesystem ${guest_domain} ${logs_fetched}`
+        echo -e "${guest_domain} ${guest_device} contains ${logs_fetched}."
+        local guest_transformed=${guest_domain//./_}
+        mkdir -p ${logs_folder}/${guest_transformed}
+        echo -e "guestfish -r -d ${guest_domain} -m ${guest_device} copy-out ${logs_fetched} ${logs_folder}/${guest_transformed}"
+        guestfish -r -d ${guest_domain} -m ${guest_device} copy-out ${logs_fetched} ${logs_folder}/${guest_transformed}
```

Comment:
which package provides this guestfish tool? Is it in our official repo?

Reply:
Just libguestfs. It is there. And installed together with SLES or qa repo